### ISSUE TITLE
[Main Tracking] App Navigation Reboot

### ISSUE THREAD

**Issue by puneetlath** [Expensify/App#11768]
**Design doc link** - https://docs.google.com/document/d/1cQ87HcDA9RrmBM4ch7PmlNi9N9ZP8abQuPF0gKYrxDE/edit

**Problem:** 

We initially implemented react-navigation in App without a proper design doc as part of the NewDot MVP. While this library came with some awesome benefits - we’ve realized over time that some of our early decisions in how we use the library have impacted performance and usability (i.e. bugs and lots of em). Our current design also deviates from our ideal design and needs to be reworked.

**Solution:** 

Let’s establish a ruleset of navigational actions a user can take and define which gestures or interactions relate to those navigational actions. Then we’ll re-implement React Navigation with those rules in mind. We will kill the drawer navigator favoring a simpler stack navigator approach. And finally, unblock the bugs and make our app navigation competitive and bug free.

**Timeline:** WhatsApp Quality Level Urgency 🔥

**Plan of Action:**

- [x] #whatsnext [link](https://expensify.slack.com/archives/CC7NECV4L/p1668651013075549) 
- [x] Send Strategy@ email 
- [x] Predesign links [1](https://expensify.slack.com/archives/C01GTK53T8Q/p1664825958481829), [2](https://expensify.slack.com/archives/C01GTK53T8Q/p1667940831213829), [3](https://expensify.slack.com/archives/C01GTK53T8Q/p1668109720905119)
- [x] HL Design Doc [link](https://docs.google.com/document/d/1cQ87HcDA9RrmBM4ch7PmlNi9N9ZP8abQuPF0gKYrxDE/edit)
- [x] Detailed Design Doc
- [x] Implement New Navigation
- [x] Unblock bugs by fixing or closing them

## Implementation PRs

- https://github.com/Expensify/App/issues/15846
- https://github.com/Expensify/App/issues/15847
- https://github.com/Expensify/App/issues/15848
- https://github.com/Expensify/App/issues/15849
- https://github.com/Expensify/App/issues/15850
- https://github.com/Expensify/App/issues/15851
- https://github.com/Expensify/App/issues/15852
- https://github.com/Expensify/App/issues/18064
- https://github.com/Expensify/App/issues/18092
- https://github.com/Expensify/App/issues/18603
- https://github.com/Expensify/App/issues/19665
- https://github.com/Expensify/App/issues/20320
- https://github.com/Expensify/App/issues/20342
- Tests:
    - https://github.com/Expensify/App/issues/15853
    - https://github.com/Expensify/App/issues/15854

## Overview
This GH will serve as the tracking issue for improving our React Navigation implementation. A bunch of issues with navigation have been found and are referenced in [this project](https://github.com/Expensify/App/projects/2).

## List of related issues
Currently fall into two major categories

###  Drawer Navigation / State

- Drawer state is unpredictable 
    - https://github.com/Expensify/App/issues/8126
    - https://github.com/Expensify/App/issues/11111
    - https://github.com/Expensify/App/issues/11961
    - https://github.com/Expensify/App/issues/11948
    - https://github.com/Expensify/App/issues/11970
    - https://github.com/Expensify/App/issues/12041
- Drawer state transition end can not be listened for - https://github.com/Expensify/App/issues/9253 
- [Home screen / MainDrawerNavigator](https://github.com/Expensify/App/blob/f95b761e117b82ecfbd28e548928591ed5749cac/src/libs/Navigation/AppNavigator/MainDrawerNavigator.js#L58-L82) is not getting route params from Report route
- Switching from large to small width screens breaks search navigation (actually the drawer is just stuck closed)
    - https://github.com/Expensify/App/issues/9734
    - https://github.com/Expensify/App/issues/9342 
- Navigating to search page using a link while logged out ends up in empty option list after signing in 
	- https://github.com/Expensify/App/issues/12717   
	- https://github.com/Expensify/App/issues/12719
	- https://github.com/Expensify/App/issues/12428
	- https://github.com/Expensify/App/issues/12616
- https://github.com/Expensify/App/issues/12128
- https://github.com/Expensify/App/issues/16319
- https://github.com/Expensify/App/issues/17147

### Browser History

- Back navigation / history stack for reports on web is [hacked in](https://github.com/Expensify/App/blob/f95b761e117b82ecfbd28e548928591ed5749cac/src/libs/Navigation/DeprecatedCustomActions.js#L95-L124) 
- mWeb navigating back exits the chat (report route not added to history
    - https://github.com/Expensify/App/issues/10507 
    - https://github.com/Expensify/App/issues/10506 
    - https://github.com/Expensify/App/issues/11372
- mWeb visual bug on Safari related to popstate event
    - https://github.com/Expensify/App/issues/7737 
- Back/Forward shortcut keys do not work
    - https://github.com/Expensify/App/issues/8657 
    - https://github.com/Expensify/App/issues/4612
    - https://github.com/Expensify/App/issues/11372
    - https://github.com/Expensify/App/issues/11420
    - https://github.com/Expensify/App/issues/11429
- After deleting stuff, navigating back can lead to a 404
    - https://github.com/Expensify/App/issues/11524
- https://github.com/Expensify/App/issues/11453
- https://github.com/Expensify/App/issues/10409
- https://github.com/Expensify/App/issues/11092
- https://github.com/Expensify/App/issues/9823
- https://github.com/Expensify/App/issues/12203
- https://github.com/Expensify/App/issues/11515
- https://github.com/Expensify/App/issues/13449
- https://github.com/Expensify/App/issues/13743
- https://github.com/Expensify/App/issues/14973
- https://github.com/Expensify/App/issues/16557

### Performance

- https://github.com/Expensify/App/issues/14036
- https://github.com/Expensify/App/issues/14173

### Misc. 
- https://github.com/Expensify/App/issues/13971
- https://github.com/Expensify/App/issues/15631
- https://github.com/Expensify/App/issues/15239
- https://github.com/Expensify/App/issues/16512

**Comment by parasharrajat** [comment]
> Get Control of React-Navigation

:fire: :fire:

**Comment by puneetlath** [comment]
Suggestion from @roryabraham: move `src/libs/Navigation` to `src/Navigation` to highlight its importance to someone coming to the repo for the first time.

**Comment by puneetlath** [comment]
Another interesting note from @roryabraham: we don't use React Navigation's [link](https://reactnavigation.org/docs/link) component.

**Comment by puneetlath** [comment]
Reached out to React Navigation library owner, but haven't heard back. Will follow up again this week as well as with perhaps some others.

**Comment by puneetlath** [comment]
Jason will be the BZ person tag teaming this with me, so assigning to him to the issue as well.

**Comment by JmillsExpensify** [comment]
@puneetlath I see you're trying to get invited to the Discord room, but is there anything else we can do to speed this part up? It's kind of a bummer that getting in contact with the maintainer is a blocker.

**Comment by puneetlath** [comment]
If I'm not able to get access by tomorrow then we'll reach out to some of our other vendors to see if they can help.

**Comment by johncschuster** [comment]
Hey @puneetlath, @marcaaron suggested we added [this issue](https://github.com/Expensify/App/issues/10506) to this larger tracking issue you're managing.

**Comment by puneetlath** [comment]
Got access to Discord today and reached out. Will see what they say and then go from there.

**Comment by JmillsExpensify** [comment]
Oh nice work!

**Comment by JmillsExpensify** [comment]
Any luck with next steps on this one?

**Comment by puneetlath** [comment]
No luck with the maintainers. Have asked margelo if they can help next.

**Comment by JmillsExpensify** [comment]
Looks like they are willing to jump in based on the Slack response.

**Comment by JmillsExpensify** [comment]
I posted about this in Slack today, but I'm also kind of thinking that we still don't have any great leads. It sounds like Margelo can likely help – and today is a holiday in Poland – so perhaps let's wait one more day for extra clarity.

**Comment by marcaaron** [comment]
I have familiarity with react navigation and bandwidth to help here if this is a good place to assist.

Does it feel like we have settled on what a new navigation design would look like based on the Slack convo [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1664825958481829)?

From where I'm sitting, it looks like we did some initial investigation and should compile that into a **Navigation 2.0** doc then once there is broad agreement on how navigation will work we can investigate what changes (if any) `react-navigation` would need? Maybe it's none or at least fewer than we are experiencing now. Bringing people in to fix things related to the drawer navigator might not make sense if our new navigation design replaces it with a stack nav.

**Comment by puneetlath** [comment]
I would very much welcome your help here @marcaaron. I have not been doing a good job at all moving this forward and am very conscious of the fact that I'm being a blocker for it. 

My plan was to write up a summary of where I think we want to go with Navigation for Margelo in the Slack channel, but I like the idea of writing it in a doc instead. That seems like a better approach.

**Comment by JmillsExpensify** [comment]
Nice, this sounds really great! Let me know if ya'll need help in any way.

**Comment by laurenreidexpensify** [comment]
@puneetlath as this is a tracking issue, does it need the Bug Label?

**Comment by JmillsExpensify** [comment]
It's fixing several bugs in the app, so I think it's appropriate.

**Comment by marcaaron** [comment]
I think it would be good to have some minimal proof of concepts together so people can actually click around and then share opinions on how things should work. I went back through some of the threads yesterday and a few points seem undecided. Will try to summarize everything by the end of today.

**Comment by marcaaron** [comment]
Spent time today on a minimal proof of concept today that tests out a basic mobile stack flow where we:

- Get rid of the `x` in right hand modal (gotta hit `<` no `x` at all)
- `<` button will either bring you to the "parent" node or the previous screen depending on context
    - The "Search" view would be in the navigation stack and a navigation like this would be possible:
        - `LHN (push) Search (push) Chat 1 (push) Chat 2 (<) Chat 1 (<) Search (<) LHN`
    - And also that if we navigate to the Settings/About screen via a Chat screen then we would start get something like this
        - `LHN (push) Chat 1 (push) [Settings, About] index 1 (<) [Settings] (<) Chat 1 (<) LHN`

I'm working on all of this in a Snack because it's easy for people to download the Expo Go app on the platform of their choice and test this stuff out for real. The UI is just placeholder stuff so you have to use your imagination a bit. Just download the app, scan the QR code, and you should be testing.

https://snack.expo.dev/@marcaaron/6e1035

Practically speaking, it is not too challenging to implement this style of view for native. So, I'm focusing on working out native/mobile web for now and we can test how it feels there. Once it's good we can think about how to get it working for desktop.

Using react navigation to use the stack based navigation hierarchy is not immediately obvious to me and may require a custom navigator so that views in the stack can be repositioned or presented however we like (assuming it's even possible). The current examples I've come across for desktop views all use a drawer navigator in the "permanent" position. But this is a good thing to seek advice on.

If anyone wants to test the web implementation you can clone this branch

https://github.com/marcaaron/navigation-test

And run `npm i && npm run web`.

It only shows a mobile web style view for now, but you can test the browser back button press vs. `<` press. In the case where we are navigating from a `Chat -> [Settings, About]` a back button press would bring you back to `Chat`, while a `<` press will bring you to `Settings`.

However, if you are on the `Settings` page another back button press brings you back to the LHN. So the browser navigation is not without bugs even when just using a stack navigator and ditching the drawer navigator 😄 

<img width="1070" alt="2022-11-03_14-25-36" src="https://user-images.githubusercontent.com/32969087/199859479-9c756c00-a9c7-4db1-b112-336d4046e091.png">

**Comment by marcaaron** [comment]
**Plan for this week:** 

_Navigation Test_

Get a nicer looking proof of concept together then deploy a mobile web test app + invite everyone to test things with Expo Go so we can see how the behavior will work on mobile web and native.

_Web Navigation with a Custom Navigator_

Do a feasibility study into how we can achieve the current “three pane view” with a stack navigator only. This is something I’ll probably bring up to Margelo, Software Mansion, and the react navigation discord to get some thoughts on before going too far down the rabbit hole.

**Comment by parasharrajat** [comment]
I would suggest testing the web and mobile web flow on the firefox browser. I have noticed that issues are easily reproducible on firefox. e.g, opening a chat via Newchat page does not always open the chat, the user is immediately navigated back to the previously active chat.

**Comment by marcaaron** [comment]
Thanks, will keep that in mind.

**Comment by JmillsExpensify** [comment]
@marcaaron is prepping a discussion for open source to keep this moving forward!

**Comment by marcaaron** [comment]
Continuing conversation here -> https://expensify.slack.com/archives/C01GTK53T8Q/p1667940831213829 🚀

**Comment by marcaaron** [comment]
I also did a study into using a [custom navigator](https://reactnavigation.org/docs/custom-navigators/) on web today with the default `StackRouter`. It works pretty good, but curious to get some more feedback on it as the main web example I've seen floating around uses a drawer navigator.

There also does seem to be some inconsistencies with browser history still when using the `StackRouter` - though it is much more predictable than the drawer routing and I think will eliminate a good amount of the bugs we're seeing. 

I think any other edge cases or "new bugs" are good thing to save for the "detailed" where we can document things we are running into and why they happen and then loop in some experts to validate the design and propose fixes.

**Comment by parasharrajat** [comment]
The advantage of a Custom navigator could be to get rid of batch navigations. Currently, our DeprecatedCustomActions trigger multiple navigation actions. 

For example, 
 
 when a user creates a new chat, the First RHN stack is popped out. then another navigation reset action is triggered to open the chat. Navigation is an expensive operation. Running two will put a load on the js thread. I think this is the reason why New Chat sometimes does not work on firefox. 

AFAIK, we can calculate the end state and fire the navigation action once in the custom navigation.

**Comment by marcaaron** [comment]
Left a [summary](https://expensify.slack.com/archives/C01GTK53T8Q/p1668109720905119) on where this is at in `#expensify-open-source` today.

It seems like the two things left to resolve are:

- Should UP and DISMISS actions add items to the navigation history?
- Should we be able to go “forward” on native?

I think once we have some clear direction there we can put this down into a doc and send out for feedback.

**Comment by marcaaron** [comment]
Still waiting for more feedback to trickle in here so I can get started on the high level doc.

**Comment by marcaaron** [comment]
Conversations are mostly wrapped up. HL Draft is in progress here -> https://docs.google.com/document/d/1cQ87HcDA9RrmBM4ch7PmlNi9N9ZP8abQuPF0gKYrxDE

**Comment by mountiny** [comment]
I have read and reviewed this Design Doc!

**Comment by puneetlath** [comment]
I have read and reviewed this Design Doc!

**Comment by trentpetty** [comment]
I have read and reviewed this Design Doc!

**Comment by madmax330** [comment]
I have read and reviewed this Design Doc!

**Comment by laurenreidexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by robertjchen** [comment]
I have read and reviewed this Design Doc!

**Comment by joaniew** [comment]
I have read and reviewed this Design Doc!

**Comment by shawnborton** [comment]
I have read and reviewed this Design Doc!

**Comment by bfitzexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by arosiclair** [comment]
I have read and reviewed this Design Doc!

**Comment by tgolen** [comment]
I have read and reviewed this Design Doc!

**Comment by francoisl** [comment]
I have read and reviewed this Design Doc!

**Comment by JmillsExpensify** [comment]
I have read and reviewed this design doc.

**Comment by JmillsExpensify** [comment]
How ironic ^^^

**Comment by Julesssss** [comment]
I have read and reviewed this Design Doc!

**Comment by iwiznia** [comment]
I have read and reviewed this Design Doc!

**Comment by neil-marcellini** [comment]
I have read and reviewed this Design Doc!

**Comment by mountiny** [comment]
We are moving to detailed section, thanks for all the great high-level reviews!

**Comment by mountiny** [comment]
Work on Detailed section continues, I am focusing on familiarizing myself with the `react-navigation` library

**Comment by tjferriss** [comment]
I have read and reviewed this Design Doc!

**Comment by sketchydroide** [comment]
I have read and reviewed this Design Doc!

**Comment by stitesExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by JmillsExpensify** [comment]
Still working through the detailed.

**Comment by marcaaron** [comment]
Had a meeting today with @mountiny and @danieldoglas to go over some current App navigation stuff. 

Here are some notes, links, and ideas for next steps:

*Meeting Notes*

- Went over existing navigation structure and some pitfalls of the drawer navigator
- Discussed potential for using the `Link` component or link props
- Touched on `DeprecatedCustomActions` and the hacks we have to support browser history and drawer state
- Custom navigator for web and how it might work

*Next steps*

- Map out some navigation flow pattern examples and test them in the proof of concept
- Put some basic wireframes together to help explain the plan for web and using the Stack navigator
- Join the React Navigation Discord and share our plan for web with the maintainers to see if they have any insights

*Links*

[https://github.com/Expensify/App/blob/main/src/libs/Navigation/Navigation.js](https://github.com/Expensify/App/blob/main/src/libs/Navigation/Navigation.js)
[https://github.com/Expensify/App/blob/main/src/libs/Navigation/linkTo.js](https://github.com/Expensify/App/blob/main/src/libs/Navigation/linkTo.js)
[https://github.com/Expensify/App/blob/main/src/libs/Navigation/linkingConfig.js](https://github.com/Expensify/App/blob/main/src/libs/Navigation/linkingConfig.js)
[https://github.com/Expensify/App/blob/main/src/libs/Navigation/DeprecatedCustomActions.js](https://github.com/Expensify/App/blob/main/src/libs/Navigation/DeprecatedCustomActions.js)

[https://reactnavigation.org/docs/stack-actions/](https://reactnavigation.org/docs/stack-actions/)
[https://reactnavigation.org/docs/navigation-state](https://reactnavigation.org/docs/navigation-state)
[https://reactnavigation.org/docs/custom-navigators](https://reactnavigation.org/docs/custom-navigators)

**Comment by trjExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by mountiny** [comment]
I will start on these 2 tasks today as well as continuing with familiarizing myself with the react-navigation:

> Put some basic wireframes together to help explain the plan for web and using the Stack navigator
> Join the React Navigation Discord and share our plan for web with the maintainers to see if they have any insights

**Comment by mountiny** [comment]
At the moment, we are working on the [POC app](https://github.com/marcaaron/navigation-test) to make sure all the actions we want work well and as expected there. We would like to add a functionality whereby resizing the web window we switch to mobile/native like navigation flawlessly.

We also reached out to SWM to get some of their experts to help validate our approach and ideas/ propose better solutions. They should come back to us next week the latest. https://expensify.slack.com/archives/C04878MDF34/p1670361091290799

**Comment by mountiny** [comment]
Latest update [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1670609759265409?thread_ts=1670271765.166879&cid=C01GTK53T8Q)

**Comment by mountiny** [comment]
Vivek continues to look into using the options to define the styles of the pages, I am unfortunately ooo sick, hoping to come back tomorrow.

**Comment by mountiny** [comment]
We had our first calls with Adam from SWM and they already proposed some ideas. We will continue to work with them tomorrow

**Comment by mountiny** [comment]
We are working closely with Adam from SWM on the proof of concept app since Thursday. They already had a chance to propose two approaches after a quick familiarization with the problem.
- One approach is to render mobile and desktop layout next to each other in one big wrapper component and then use stying to either show or hide one of the layouts based on what sizing we want
    - this approach should be fine in terms of the responsiveness requirement, but we can anticipate issues with native animations/transitions as those might not be that easy to achieve and their performance would not be the same as using a classic native stack navigator
- Another option is the one we hoped for from the beginning and that is to switch between the navigators as we resizes by copying the navigation state/ history between the navigators. This should be possible and Adam is researching how to make the transformation of the state between the navigators
    - Caveat of this approach is that we will have to dismount and mount the screens/ components again as we switch between the navigators, this leads to any component state being lost. We have discussed this in slack with Marc and Vivek and I think we can get around this by storing whatever important state we dont want to loose (long bank information forms or similar kind of inputs) in Onyx
    - This approach is favourable because it allows us to use the Native Stack navigator for mobile which is the navigator using native api layers for navigations and hence the performance/ feel/ look is what is closest to the native apps and what we want to achieve

#### Next steps

I will be in touch with Adam to confirm we can make the transformation between the two navigators without loosing any history or other detrimental effects on the UX.

I will update this PR which simplifies navigation to nested stack https://github.com/marcaaron/navigation-test/pull/5

**Comment by mountiny** [comment]
Keeping the updates in [this thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1671463739572979) this week. Adam has shared a [WIP branch](https://github.com/marcaaron/navigation-test/pull/8) where he already has some parts of the responsiveness which is promising and we can build on top of that. 

I will try to do some testing tomorrow and sync with Adam to see where we are at and what problems he has faced with.

**Comment by mountiny** [comment]
Still ironing out the details of the transformation of the navigator state, prepping for deep linking as well. 

Next steps are to ensure the browser history (back button logic in browser) work after the transition from native (small screen) to web layout.

**Comment by JmillsExpensify** [comment]
Removing from WAQ per [this Slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1672088203968479).

**Comment by mountiny** [comment]
The hustle slowed down a bit over the Holidays due to time off, however, Adam has posted an update [just now](https://expensify.slack.com/archives/C04878MDF34/p1672335962646979). We will try to catch up tomorrow at some point to chat about the direction SWM wants to explore as it seems like the cross-platform aspect (and supporting multiple layouts/ navigators) will have to be hacky.

I will provide hopefully more detailed update once we catch up

**Comment by JmillsExpensify** [comment]
Should we go ahead and create a project for this initiative? It's not super critical at the moment, because we're still in the research phase. I'm mainly thinking about how we keep the right visibility on the project for right now and when we get to implementation.

**Comment by mountiny** [comment]
We are discussing whats the best way to move forward with this project and its management. By the SWM's research and removing from WAQ we got to a cross-road where we will ideally need to do other predesign meeting to see how to move forwards.

@marcaaron has nicely summed it up [here](https://expensify.slack.com/archives/C04878MDF34/p1672435517754829?thread_ts=1672335962.646979&cid=C04878MDF34)

There are some limitation of the react-navigation which is not made to be cross-platform so we will inevitably need to have hacks to support web and mobile at the same time. SWM works on new library with cross-platform in its core, however, it is still early and in "beta" phase. We need to make sure we all align on whats the best way forwards.

**Comment by mountiny** [comment]
Latest update [in Slack](https://expensify.slack.com/archives/C04878MDF34/p1672837290586499?thread_ts=1672776718.586399&cid=C04878MDF34).

I will post a larger update in Slack to get everyone on board with the current plan we have or if people prefers different directions.

**Comment by mountiny** [comment]
Posted an update and started a discussion in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q/p1672930325359569).

**Comment by mountiny** [comment]
Adam will hopefully be back in the game today after being sick last week. I will try to do some more stuff in the design doc which is not hold on the code changes.

**Comment by mountiny** [comment]
Brought one question to Slack so we can get opinions of other people on this potential change in web Back button navigation pattern https://expensify.slack.com/archives/C01GTK53T8Q/p1673363145534749

**Comment by mountiny** [comment]
Had a good discussion in the slack threads linked above. We will take our time to try to achieve cross-platform consistency as much as possible. It will most likely require some further pre-design as there are some flows which are unclear only from the high-level now.

For example, once we allow tablets to use landscape, we have this scenario (and describes what the web app should do in this layout as well). In Native Android app, on tablet in landscape, user would have this layout:
<img width="1478" alt="image" src="https://user-images.githubusercontent.com/36083550/212320293-569fdfc5-c58e-44ba-920e-db308eac6dd0.png">
Then:
- user navigates to chat A using the LHN
- user navigates to chat B using the LHN
- user navigates to chat C using the LHN
- user clicks the Android tablet back button

What does android back button does in this case?

There would most likely be more usecases we need to address and the more examples we go through and agree on how they will work.

**Comment by marcaaron** [comment]
As per the initial predesign conversation navigating to chats is an "additive" action. A common request for people when using web is that navigating to chats should add items to the browser history a.k.a. back button takes you to a previous chat. So, we would do the same on Android? I don't think it matters that the LHN is visible here. One of the core parts of our design discussion was that navigating to a chat would push it onto a "stack".

**Comment by mountiny** [comment]
> As per the initial predesign conversation navigating to chats is an "additive" action. A common request for people when using web is that navigating to chats should add items to the browser history a.k.a. back button takes you to a previous chat. So, we would do the same on Android? I don't think it matters that the LHN is visible here. One of the core parts of our design discussion was that navigating to a chat would push it onto a "stack".

Thanks for clarifying, I will try to add these web-layout examples to the design doc, ideally as some visual examples. When chatting about these, its showing that people find it hard to be sure whats the expected behaviour with the web layout navigation. 

The mobile navigation is intuitive, but since the layout is different than on desktop, saying the back button will do the same as on mobile is not clear since that looks differently.

Making a set of examples and edge cases listed in the testing section of the design doc will be also helpful for the POC since we can go through all of those flows and ensure they behave as we want.

**Comment by marcaaron** [comment]
>I will try to add these web-layout examples to the design doc, ideally as some visual examples. When chatting about these, its showing that people find it hard to be sure whats the expected behaviour with the web layout navigation.

Thanks, that would be really great!

>The mobile navigation is intuitive, but since the layout is different than on desktop, saying the back button will do the same as on mobile is not clear since that looks differently.

Can you rephrase? I did not understand this part.

>Making a set of examples and edge cases listed in the testing section of the design doc will be also helpful for the POC since we can go through all of those flows and ensure they behave as we want.

Are we still working on a POC? I'm not too sure after [this comment](https://expensify.slack.com/archives/C01GTK53T8Q/p1673545548192569?thread_ts=1673363183.637219&cid=C01GTK53T8Q). Or maybe you are referring to some distant POC of the future and not the one that we have been working on.

Let's sync up so we are clear on the direction post that Slack conversation.

**Comment by mountiny** [comment]
> The mobile navigation is intuitive, but since the layout is different than on desktop, saying the back button will do the same as on mobile is not clear since that looks differently.

Basically we have defined the navigation actions in general and then showcased how this works on mobile layout. Mobile layout is different and includes the LHN route (root showing the list of chats). This does not exist in desktop layout hence we can never navigate back to this exact same root and root of desktop always shows _some_ chat report as well.

This means that on mobile, if you navigate to some chat, then if you want to move to some other chat, you need to pop that Chat screen first to get to LHN so you can choose different chat. This will mean user would exit the App after two back button click from the chat report. 

On desktop, you dont need to go back before navigating to another report, which means that chat screens wont be removed from the stack so back button will go through the chat screens as you expect on the browser.

The actions are same under the hood of course, but I think it will be helpful to just put examples down for the desktop layout as well so people can ensure they understand the expected patterns correctly

**Comment by marcaaron** [comment]
>This means that on mobile, if you navigate to some chat, then if you want to move to some other chat, you need to pop that Chat screen first to get to LHN so you can choose different chat. This will mean user would exit the App after two back button click from the chat report.

**My thinking here:** You can navigate from one chat to another chat via a deep link so the behavior is not really "different" depending on the platform. The main difference is that on a desktop/web/tablet view you have basically a list of "deep links" to new chats that would get "pushed" onto the stack. 

I think the most important thing is to focus on: 

How does this change someone's expectations?

Why does it matter?

- We get a "web-like" navigation for reports because we are pushing the reports on web (correct browser back button behavior)
- The native behavior stays "native-y" because you are typically going "back" before you are going forward again - though there's also nothing that stops you from pushing new chats (which remember will be "threads" in the future).

**Comment by marcaaron** [comment]
>This will mean user would exit the App after two back button click from the chat report.

Sorry, it's not clear to me what problem we are talking about. I would suggest we move this conversation to a private DM or public Slack thread and use the main tracking issue to summarize instead of use it to discuss implementation details.

**Comment by marcaaron** [comment]
Thanks, sounds like we agree we can use this issue mainly to update people on how the project going at a high level moving forward.

**Comment by mountiny** [comment]
Posted a quick update in open-source channel [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1674079827449979).

**Comment by mountiny** [comment]
The state at the moment:
- the current proposed [branch](https://github.com/marcaaron/navigation-test/pull/8) in the POC app is promising there is just one point compared to high-level design, which is being discussed and will probably require further changes.
- Adam is working on cleaning up the PR and writing up notes which will explain the changes in plain English (which will be needed for design doc as well)
- Adam has written up a summary of the previous attempts and approaches to solve this problem (design doc's alternate solutions basically)

Next steps:
- figure out how to handle [this case](https://github.com/marcaaron/navigation-test/pull/8#discussion_r1083827779)

**Comment by mountiny** [comment]
Latest update from SWM is [here](https://expensify.slack.com/archives/C04878MDF34/p1674587622289449).

- Vivek pointed out one more bug
- Adam seems to have fixed it now
- We are going to test the POC in various cases and we can see if nothing breaks

**Comment by mountiny** [comment]
Daily Update:
- Adam has made a good summary and explanation of the changes in [the POC](https://github.com/marcaaron/navigation-test/pull/8) which will yield the behaviour we need
- we are still not super clear on whats happening when we switch between the mobile and the desktop navigator and we would like to get to the bottom of this issue. Our solution works, but it cures the symptoms and we are not sure exactly about the root cause.

**Comment by mountiny** [comment]
Daily Update:
- latest message from SWM can be found [here](https://expensify.slack.com/archives/C04878MDF34/p1674757019903629).
- we are now mainly exploring if we can find a way without patching react-navigation which would be the ideal solution. Adam has tried a different approach which needs to be tested again: https://github.com/marcaaron/navigation-test/pull/8
- we are now confident with the general direction we have, but taking time to ensure its the best one we could come up with, I believe that if everything goes well, we could do technical pre-design in #expensify-open-source next week

**Comment by marcaaron** [comment]
Last I heard, we are engaging some `react-navigation` folks to help with the proposal. Excited to see the results of that conversation.

**Comment by mountiny** [comment]
Daily Update:
- We are polishing the proposed approach before we bring this solution to detailed predesign
- Working with Adam to write the predesign update on his own

**Comment by mountiny** [comment]
Daily Update:
- still testing and polishing the POC app before we bring this solution to detailed predesign

**Comment by mountiny** [comment]
Daily Update:
- animations in desktop seems to be fine
- we will share update/ plan for the detailed section in open-source channel on Monday and kick-off the detailed design doc work if there will be no pushback to the plan

**Comment by mountiny** [comment]
Sharing a plan for the detailed section today and planning to start work on the detailed design doc section with Adam and Vivek

**Comment by mountiny** [comment]
Had a call with Adam today about design doc and how the detailed section should look like, working on the detailed section is his main focus now and me and Vivek will support him

**Comment by mountiny** [comment]
Adam has made a good progress with adding the skeleton to the detailed design doc section today. We keep working on that

**Comment by mountiny** [comment]
Detailed section work continues, Adam is now familiarizing himself with NewDot a bit more so he can get the changes suited for the App better.

**Comment by mountiny** [comment]
Working on the design doc

**Comment by mountiny** [comment]
- on Monday I had a sync with Adam and left review on the design doc
- we had created some first outline/structure is going to follow 
- Adam is ooo today

**Comment by marcaaron** [comment]
Detailed doc is still in progress here.

**Comment by mountiny** [comment]
The detailed section is starting to get great shape, Adam has addressed couple rounds of my reviews and now I am asking Marc and Vivek to give their reviews too to improve the detailed section even further 🙌

**Comment by techievivek** [comment]
We're actually making some awesome progress on the app navigation reboot project. Marc gave it a review last Friday, and I went over it again on Sunday, and I've got to say, it's looking pretty good overall. We're getting pretty close to having the detailed section ready, so let's keep up the great work! 🙌 

Vit is ooo today and tomorrow.

**Comment by JmillsExpensify** [comment]
Wow nice work!

**Comment by mountiny** [comment]
Adam is further polishing the design doc, we will try to give it another review to provide feedback and any support he might need.

**Comment by mountiny** [comment]
Adam found out one issue we faced should be fixed with the latest react-navigation version. He is now trying to figure out how to add animations for the right hand modal opening and closing.

**Comment by mountiny** [comment]
Asking for some early reviews, aiming to send the detailed section out for a review early next week

**Comment by JmillsExpensify** [comment]
Based on the last comment, it's possible a design doc could be out for review this week.

**Comment by mountiny** [comment]
finishing last touches and gonna send the design doc out for a review today/tomorrow

**Comment by MelvinBot** [comment]
:wave: Hello Generalist Track Team - you have been assigned to review this **Detailed** Design Doc. Please, treat this as a priority. Review [this Stack Overflow](https://stackoverflow.com/c/expensify/questions/5838) for some tips on reviewing a design doc. Once you are done, simply press the `Add "Reviewed Doc" comment` button in the right hand side K2 pannel or follow [these](https://stackoverflow.com/c/expensify/questions/13713) instructions.

---
- `@arosiclair` (Expensifier / Graduate) - https://github.com/Expensify/App/issues/15523
- `@alexpensify` (Expensifier / Graduate) - https://github.com/Expensify/App/issues/15524
- `@mateocole` (Project Manager) - https://github.com/Expensify/App/issues/15525
- `@tjferriss` (Project Manager) - https://github.com/Expensify/App/issues/15526
- `@robertjchen` (Product Manager / Generalist + Engineer) - https://github.com/Expensify/App/issues/15527
- `@nkuoch` (Product Manager / Generalist + Engineer) - https://github.com/Expensify/App/issues/15528
- `@cead22` (Product Manager / Generalist + Engineer) - https://github.com/Expensify/App/issues/15529
- `@AndrewGable` (Product Manager / Generalist + Engineer) - https://github.com/Expensify/App/issues/15530
- `@laurenreidexpensify` (Product Manager / Generalist + Non-Engineer) - https://github.com/Expensify/App/issues/15531
- `@shawnborton` (Product Manager / Generalist + Non-Engineer) - https://github.com/Expensify/App/issues/15532
- `@RobertLadue` (Product Manager / Generalist + Non-Engineer) - https://github.com/Expensify/App/issues/15533
- `@anmurali` (Product Manager / Generalist + Non-Engineer) - https://github.com/Expensify/App/issues/15534

**Comment by mountiny** [comment]
Here is the design doc link https://docs.google.com/document/d/1cQ87HcDA9RrmBM4ch7PmlNi9N9ZP8abQuPF0gKYrxDE/edit#heading=h.aj1p1g4afz1y

Thanks for a review in advance!

**Comment by roryabraham** [comment]
I have read and reviewed this Design Doc!

**Comment by luacmartins** [comment]
I have read and reviewed this Design Doc!

**Comment by shawnborton** [comment]
I have read and reviewed this Design Doc!

**Comment by mountiny** [comment]
Gonna make this weekly as we are getting through the review comments. Got some great ones so far!

**Comment by arosiclair** [comment]
I have read and reviewed this Design Doc!

**Comment by AndrewGable** [comment]
I have read and reviewed this Design Doc!

**Comment by tgolen** [comment]
I have read and reviewed this Design Doc!

**Comment by nkuoch** [comment]
I have read and reviewed this Design Doc!

**Comment by alexpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by mountiny** [comment]
Got in a good number of review, missing 4 more product managers/ generalist reviews so since we are hoping to be able to implement this during ECX, I have bumped the reviews to daily, thanks!

**Comment by tjferriss** [comment]
I have read and reviewed this Design Doc!

**Comment by robertjchen** [comment]
I have read and reviewed this Design Doc!

**Comment by mountiny** [comment]
Resolved all the existing comments, we need 3 more reviews from Product managers or Generalists to be able to start the implementation. Trying to get those before ECX

**Comment by JmillsExpensify** [comment]
I have read and reviewed this Design Doc!

**Comment by RobertLadue** [comment]
I have read and reviewed this Design Doc!

**Comment by mountiny** [comment]
Thanks everyone for your reviews, we have received the required number of reviews so we are moving into implementation and we will work with Adam from SWM during ExpensiconX. I have added the issues to the issue body but also adding them here for clarity

## Implementation PRs

- https://github.com/Expensify/App/issues/15846
- https://github.com/Expensify/App/issues/15847
- https://github.com/Expensify/App/issues/15848
- https://github.com/Expensify/App/issues/15849
- https://github.com/Expensify/App/issues/15850
- https://github.com/Expensify/App/issues/15851
- https://github.com/Expensify/App/issues/15852
- Tests:
    - https://github.com/Expensify/App/issues/15853
    - https://github.com/Expensify/App/issues/15854

**Comment by mallenexpensify** [comment]
QQ - Should I put this on hold pending Nav reboot?
- https://github.com/Expensify/App/issues/15785

**Comment by marcaaron** [comment]
@mallenexpensify Would love to keep that discussion on the related issue. I'll leave a comment there in a moment. Thanks!

**Comment by mountiny** [comment]
We have merged a first PR to the `navigation-branch` last week and Adam continues to make progress on this

**Comment by JmillsExpensify** [comment]
@mountiny Can you let me know when the first round of PRs hit staging? I'd like to re-test a bunch of issues we've kept on hold and get ahead closing the ones that are already fixed as a result of this project.

**Comment by mountiny** [comment]
Will do!

At the moment, we have [this draft PR](https://github.com/Expensify/App/pull/16581) which addressed 3 issues actually as that was easier way to handle this. We got 3 C+ on it to review and test and help Adam find any missing spots. The progress is being made although it takes a bit as there is lots of report and drawer related logic which needs to be refactored/removed.

**Comment by cead22** [comment]
I have read and reviewed this Design Doc!

**Comment by cead22** [comment]
I have read and reviewed this Design Doc!

**Comment by adamgrzybowski** [comment]
> Are we going to remove `React-Freeze`
> 
> https://github.com/Expensify/App/blob/4ee21e95a38858df28e46caf54dc97e1bca9bc7a/src/pages/home/ReportScreen.js#L233
> 
> as part of this reboot?

Do you see any reason to remove it from the codebase? Currently, we don't have plans to remove it. But we can always discuss it

**Comment by adamgrzybowski** [comment]
That sound like a good idea! Thanks! 🙇

**Comment by mountiny** [comment]
Update:

Adam is working through removing the drawer from the navigation, this inevitably shows some unexpected bugs, but with collaboration of C+ we are finding those and they can fix them once the steps are reproducible. Adam also got help of Mariusz and possible Wojtek soon to help move this project forwards towards completion faster given its complexity.

**Comment by mountiny** [comment]
Implementation still going strong. C+ has helped a lot with testing and identifying some bugs, we are now focusing on getting the first PR which addresses 3 issues merged in the `navigation-refactor` branch and move onto the other issues.

**Comment by mountiny** [comment]
@adamgrzybowski has identified a small bug in react-navigation code and together with Wojtek they will try to get that fixed upstream in react-navigation library. The [main PR](https://github.com/Expensify/App/pull/16581) seems to be close to being merged to the navigation-refactor branch, there are still automated tests failing because of change to the architecture of the navigation but hopefully those can be resolved soon.

The following PRs should be smaller in scope so those should move forwards faster.

**Comment by mountiny** [comment]
I have merged the main PR to the `navigation-refactor` branch:

- https://github.com/Expensify/App/issues/15848
- https://github.com/Expensify/App/issues/15851
- https://github.com/Expensify/App/issues/15852

These three issues then can be closed, one of them is still open to handle payments to C+.

We are onto the other issues, which should hopefully have smaller PRs which are easier to handle so we are a bit closer to our goal here 🎉 Great job @adamgrzybowski @staszekscp, lets keep this up!

**Comment by roryabraham** [comment]
Now that the Drawer navigator has been removed, am I correct in thinking that there are a bunch of PRs that can come off HOLD?

**Comment by mountiny** [comment]
> Now that the Drawer navigator has been removed, am I correct in thinking that there are a bunch of PRs that can come off HOLD?

It has not been merged to main, we are going to implement all the improvements into the `navigation-refactor` branch. Merging this has unheld the remaining 3 issues hence PRs for those, but we will have to wait for all of this to be ready to merge into main after we give this all a good test on physical devices from `navigation-refactor`

**Comment by JmillsExpensify** [comment]
Wow this is amazing! Regarding the held issues, which at one point was like 1/3 of all issues held in the repo, I think we need to have a plan for how we get visibility and urgency on those once they are indeed off hold? Perhaps when we're readying, I can post in open source and we can align on next steps? I think we need to do something different and higher profile given just how many issues this is.

**Comment by mountiny** [comment]
We are down to two issues:
- https://github.com/Expensify/App/issues/15849
- https://github.com/Expensify/App/issues/15850 - should be easy

And fixing some bugs we will discover with the new architecture when testing

The main challenge in implementing the UP functionality now as we discovered we will need some react-navigation changes and Adam with his team is trying to get those to react-navigation core.

Then I think we should first address the Drawer related bugs we saw as those should hopefully be resolved. Then we can take on the rest in not a specific order. I think ideally, one this is deployed to staging, we can ask QA to run those HELD issues as KI and confirm if they can reproduce them or not. There is a high chance we will also get some new issues too.

**Comment by marcaaron** [comment]
>The main challenge in implementing the UP functionality now as we discovered we will need some react-navigation changes and Adam with his team is trying to get those to react-navigation core.

It sounded like maybe the problem was smaller than we originally thought ([see thread here](https://expensify.slack.com/archives/C01GTK53T8Q/p1682017824094319?thread_ts=1681989386.817059&cid=C01GTK53T8Q)).

**Comment by marcaaron** [comment]
>The fact is that most of these changes will fix more general problems not related only to the new UP implementation. For example unnecessary reloads of the report screen under the overlay.
>These bugs in react-navigation just happen to be more visible now while we are working on the UP implementation because we have to take a closer look at the internal react-navigation history state in the main application with a more complex structure than the one present in the POC.

Ok it sounds good to me. Less bugs = better. Mostly with the questions I am trying to understand what exactly we need to change and whether it needs to block anything. That will help us understand better which bugs we are fixing specifically and what the priority is.

**Comment by mountiny** [comment]
A PR with the UP button implementation [is being reviewed and tested](https://github.com/Expensify/App/pull/18402). Then the next steps will be to test the [navigation reboot PR](https://github.com/Expensify/App/pull/18605) and polish any main bugs we might face and which would make using the app challenging. Once Applause wont find new regressions on the PR we will merge it to main

**Comment by JmillsExpensify** [comment]
Weekly updates continue to happen in Slack. We've been focused on external initiatives over the last week.

**Comment by mountiny** [comment]
We are down to the main PR https://github.com/Expensify/App/pull/19263 which once merge will switch the App to using Stack navigation. At the moment, Applause is going through the regression tests and they will come back to us with issues they found and we can see if any of them are blockers.

**Comment by mountiny** [comment]
Update:

the last PR is very close https://github.com/Expensify/App/pull/19263, full regression has been done on all the platforms and around 10 issues have been found and they all should have been addressed now.

There is just one thing we are trying to find solution for now before merging this to main:
- Since we are always pushing new pages to the stack when navigating between reports, on desktop/web the nuber of pages in stack can grow large over time. Even when using the freeze on the report screens, after a day of use the performance of the app with many reports in the stack got slow.

**Comment by mountiny** [comment]
The PR https://github.com/Expensify/App/pull/19263 is basically ready to be merged, however, we got an important feature which add support for unvalidated emails here https://github.com/Expensify/App/pull/18293 and it needs to work for Friday and we cannot test this with the navigation refactor until its merged.

Hence the plan is to continue polishing the branch, add performance improvements and then merge the refactor at the end fo the week so it cannot potentially influence the RenderATL presentation

**Comment by JmillsExpensify** [comment]
This is [on staging](https://expensify.slack.com/archives/C01GTK53T8Q/p1686087426497529). Let the real-world testing begin!

**Comment by mountiny** [comment]
The main PR just got merged and stack navigation is available in Staging. I have compiled a list of some clean up and follow up tasks in [this issue](https://github.com/Expensify/App/issues/20320), working on adding a documentation to the main about the navigation.

**Comment by roryabraham** [comment]
Seeking guidance on whether this is now expected behavior: https://github.com/Expensify/App/issues/20370

It seems like the answer is probably "yes" given the new stack navigator?

**Comment by hayata-suenaga** [comment]
I have read and reviewed this Design Doc!

**Comment by mountiny** [comment]
The migration has been done at the end of the last week and it went surprisingly smooth. I am not tracking issues related to the new navigation in this public project https://github.com/orgs/Expensify/projects/37 and trying to iron them out with SWM.

With Jason, once we get a bit of time we will go through all the issues on hold and see which are fixed and which need to be addressed still

**Comment by mallenexpensify** [comment]
I put the below on hold, pending the reboot. If anyone disagrees, please comment on the issue
- https://github.com/Expensify/App/issues/21229

**Comment by mountiny** [comment]
I will check it out, the refactor is deployed now and we are going through some follow up issues and we will go though all the issues on hold to test them and close issues we cannot repro

**Comment by mountiny** [comment]
Added all the relevant issues to the https://github.com/orgs/Expensify/projects/39 and to those we need to go through and confirm if they are fixed are added to the TODO column.

**Comment by mountiny** [comment]
I will try to close this out this week and send a wrap up!

**Comment by JmillsExpensify** [comment]
And I am still working on testing the linked issue. Should be able to finish it up this week.

**Comment by mountiny** [comment]
Going to send out project wrap up

**Comment by mountiny** [comment]
🗺️🧭 It’s time for the Navigation Reboot wrap-up 🎉

Its been almost 9 months since we embarked on the journey to update out navigation concepts in newDot to have more reliable and predictable navigation pattern across all our supported platforms. And today, this issue can be finally closed 🎯

This was one of the first projects where we have fully outsourced to an external agency, Software Mansion. I am also tagging @adamgrzybowski  and @WoLewicki  who have done amazing job navigating unknown codebase, learning our design doc process and delivering this solution with care and urgency. Thank you guys 👏


**What went well**

- We have closed over 28 bugs (more to be closed but we haven't got through them yet) which were on hold for this project. You can see them in [this Github issue](https://github.com/orgs/Expensify/projects/39) 💪

- We now have way more predictable navigation system which is easier to use and has less custom actions (which had steep learning curve for anyone new to the the codebase)

- This project has shown we can effectively expand our teams through the expert agencies if we can work using the “buddy” system and find a project that suits them. In this case, basically 100% of the implementation has been done in the App which was ideal.

- It has been a great learning experience on how to work with the expert agencies, I haven’t got to this yet, but I would love to start some SO where we can share best practices on how to work as a “buddy” and other engineers can share their experiences there too.

- The detailed section of the design doc has been very much written by the external agency and I have only helped with minor things and resolving the threads. The doc has got great feedback, hopefully we can do more docs like this with SWM 🎀. This quality has helped to save internal engineering time.

 - The implementation has went quite smooth, it was a major change in the codebase, but there were no major regressions or blockers and we managed to address any issues after the “switch" in timely manner.


**What we learned**

- Lots about how to work with the external agencies. This is probably not surprising, but to allow them to work as efficiently as possible, the buddy should be proactive and be available as much as possible whenever they get stuck/ have questions. Given they usually have this single project as their main focus, if they are stuck on something, they will be wasting their time until we help them to get unstuck.

- It can take a lot of time to manage such project due to the above and being in similar timezone helps a lot to keep it going faster (move to Europe) but I certainly believe these projects can unlock so much productivity and high quality front-end code that we would hardly achieve in the same timeframe without the experts.

- Using Github Projects is quite handy for managing the open source projects. It would be even better if we could let the expert contributors move the card within the projects on their own, but we would most likely have to make them all part of our Organization.


**What we could have done better**

- This one could be in the previous category but: whenever the project is discussing a bit more abstract or ambivalent terms and concepts, its crucial to keep the conversation grounded in written down and well defined terminology. Always make sure to come back to the terminology section whenever the conversation deviates or you have a feeling someone is using some term with different meaning than was agreed on. We have had lots of discussions on what should the Up, Back, Push, etc actions do within the App and often people got confused about what they actually mean and wasted a lot of time as people were not on the same page with these terms important for the project. 
    
	- Let’s make sure that whenever there are some non-trival terms in some discussions you are having in public rooms, assume people know nothing and take your time explaining the terms to get everyone on the same page to get the most out of the conversation.

- The project could have been completed faster especially if we would have repeatedly not run into confusing situations when discussing the design because of the above mentioned problem. 

**What’s next**

We have ironed out most of the follow up bugs we have faced, but we are not ending here. I am tracking all of the follow up issues in [this github project](https://github.com/orgs/Expensify/projects/37/views/1). There are 2 bigger issues addressed:

	- Resizing the browser window from wide to narrow layout or viceversa make the components to loose their state - Adam already has [an active PR](https://github.com/Expensify/App/pull/22437) to address this by changing the layout to style based layout so we dont have to remount the navigators and state of components is not lost 🎉
	
	- Trying to fix the browser back button behaviour. Browsers intentionally make it hard to mess with its internal history/ state. This makes it hard for us in many situations, but for example [here](https://github.com/Expensify/App/issues/22101), when user clicks to leave a room or a thread, they are navigated to some other page, but when they click browser back, this naturally takes them to the previous URL, which is the thread they have just left. Its not a huge problem but[ SWM is working](https://expensify.slack.com/archives/C04878MDF34/p1689872643785089?thread_ts=1689871481.652229&cid=C04878MDF34) on potential update to react-navigation which would allow us to handle this better.


Thanks everyone! Onwards and upwards! 🚀

Vit, Adam and Wojti from SWM, and also Marc and Vivek from the high level stages

**Comment by mallenexpensify** [comment]
Addressing @adeel0202's above comment [here](https://github.com/Expensify/App/issues/16512#issuecomment-1661190611)
